components/infobox/wikis/starcraft/infobox_person_player_custom.lua

```diff
@@ -119,71 +103,57 @@ function CustomInjector:addCustomCells(widgets)
 	local yearsActive = Logic.readBool(_args.enableYearsActive)
 		and Namespace.isMain() and CustomPlayer._getMatchupData() or nil
 
-	local currentYearEarnings = _earningsGlobal[tostring(_CURRENT_YEAR)]
+	local currentYearEarnings = _earningsGlobal[tostring(CURRENT_YEAR)]
 	if currentYearEarnings then
 		currentYearEarnings = Math.round{currentYearEarnings}
 		currentYearEarnings = '$' .. mw.language.new('en'):formatNum(currentYearEarnings)
 	end
 
 	return {
 		Cell{
-			name = 'Approx. Winnings ' .. _CURRENT_YEAR,
+			name = 'Approx. Winnings ' .. CURRENT_YEAR,
 			content = {currentYearEarnings}
 		},
 		Cell{name = 'Years active', content = {yearsActive}}
 	}
 end
 
 function CustomPlayer.nameDisplay()
-	CustomPlayer._getRaceData(_args.race)
-	local raceIcon = RaceIcon.getNormalIcon{_raceData.race}
+	local factions = CustomPlayer._readFactions(_args.race or Faction.defaultFaction)
+
+	local raceIcons = table.concat(Array.map(factions, function(faction)
+		return Faction.Icon{faction = faction, size = 'medium'}
+	end))
+
 	local name = _args.id or _player.pagename
 
-	return raceIcon .. '&nbsp;' .. name
+	return raceIcons .. '&nbsp;' .. name
 end
 
-function CustomPlayer._getRaceData(race, asCategory)
-	race = string.lower(race or _UNKNOWN_RACE)
-	race = CleanRace[race] or race
-	local raceTable = _RACE_DATA[race]
-
-	local faction, faction2
-	if race == _RACE_ALL_SHORT then
-		faction = _RACE_ALL
-	else
-		faction = (raceTable or {})[1]
-		faction2 = (raceTable or {})[2]
-	end
+function CustomPlayer._getRaceDisplay(race, asCategory)
+	local factions = Array.map(CustomPlayer._readFactions(race), Faction.toName)
 
-	local display
-	if not raceTable and race ~= _UNKNOWN_RACE then
-		display = '[[Category:InfoboxRaceError]]<strong class="error">' ..
-			mw.text.nowiki('Error: Invalid Race') .. '</strong>'
-	elseif Table.isNotEmpty(raceTable) then
+	return table.concat(Array.map(factions or {}, function(faction)
 		if asCategory then
-			for raceIndex, raceValue in ipairs(raceTable) do
-				raceTable[raceIndex] = ':Category:' .. raceValue .. ' Players|' .. raceValue .. ']]'
-					.. '[[Category:' .. raceValue .. ' Players'
-			end
+			return '[[:Category:' .. faction .. ' Players|' .. faction .. ']]'
 		end
-		display = '[[' .. table.concat(raceTable, ']],&nbsp;[[') .. ']]'
-	end
-
-	_raceData = {
-		race = race,
-		faction = faction or '',
-		faction2 = faction2 or '',
-		display = display,
-	}
+		return '[[' .. faction .. ']]'
+	end) or {}, ',&nbsp;')
+end
 
-	return display
+function CustomPlayer._readFactions(input)
```

Comment:
This called 4 or 5 times in this module, but returns the same thing every time. I guess this module was built before we decided to have a separate input processing stage?